pytradfri/resource.py

```diff
@@ -3,41 +3,130 @@
 
 from abc import abstractmethod
 from datetime import datetime
-from typing import Any, Callable, Dict, List, Union
+from typing import Any, Callable, Dict, List, Optional, Union
+
+from pydantic import BaseModel, Field
 
 from .command import Command, TypeProcessResultCb
-from .const import ATTR_CREATED_AT, ATTR_ID, ATTR_NAME
-from .typing import ApiResourceResponse
+from .const import (
+    ATTR_APPLICATION_TYPE,
+    ATTR_CREATED_AT,
+    ATTR_DEVICE_BATTERY,
+    ATTR_DEVICE_FIRMWARE_VERSION,
+    ATTR_DEVICE_INFO,
+    ATTR_DEVICE_MANUFACTURER,
+    ATTR_DEVICE_MODEL_NUMBER,
+    ATTR_DEVICE_POWER_SOURCE,
+    ATTR_DEVICE_SERIAL,
+    ATTR_ID,
+    ATTR_LAST_SEEN,
+    ATTR_LIGHT_CONTROL,
+    ATTR_NAME,
+    ATTR_OTA_UPDATE_STATE,
+    ATTR_REACHABLE_STATE,
+    ATTR_START_BLINDS,
+    ATTR_SWITCH_PLUG,
+    ROOT_AIR_PURIFIER,
+    ROOT_SIGNAL_REPEATER,
+)
+from .typing import AirPurifierResponse, BlindResponse
 
 # type alias
 TypeRaw = Dict[str, Union[str, int, List[Dict[str, Union[str, int]]]]]
 TypeRawSimple = Dict[str, Union[str, int]]
 TypeRawList = Dict[str, List[Dict[str, Union[str, int]]]]
 
 
+class ApiResourceResponse(BaseModel):
+    """Represent a resource response."""
+
+    id: int = Field(alias=ATTR_ID)
+    name: str = Field(alias=ATTR_NAME)
+    created_at: Optional[int] = Field(alias=ATTR_CREATED_AT)
+    ota_update_state: int | None = Field(alias=ATTR_OTA_UPDATE_STATE)
```

Comment:
We can't use the `|` syntax to mark things optional in pydantic. We need to use `Optional`.

Reply:
Note that `ota_update_state` is a new attribute.